torch/csrc/autograd/variable.cpp

```diff
@@ -164,6 +164,13 @@ namespace impl {
     auto* meta = materialize_autograd_meta(self);
     meta->grad_fn_ = std::move(edge.function);
     meta->output_nr_ = edge.input_nr;
+    // For views, make sure this new grad_fn_ is not overwritten unless it is necessary
```

Comment:
I could believe that this is correct, but I don't see the chain of causality where setting `attr_version` prevents the `grad_fn` from being overwritten. It would be helpful to have a code pointer to the point where the overwrite happens.

Reply:
It is in the `VariableHooks::grad_fn()` in this same file, where we check `diff_view_meta->attr_version != current_version` to know if the graph should be rewritten or not.